devicetypes/sky-nie/min-smart-plug.src/min-smart-plug.groovy

```diff
@@ -0,0 +1,530 @@
+
+        if (btnVal) {
+            sendButtonEvent(btnVal)
+        }
+    }
+    return []
+}
+
+private sendButtonEvent(value) {
+    if (childDevices) {
+        childDevices[0].sendEvent(name: "button", value: value, data:[buttonNumber: 1], isStateChange: true)
+    }
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+    logDebug "Unhandled zwaveEvent: $cmd"
+    return []
+}
+
+private updateSyncingStatus() {
+    sendEventIfNew("syncStatus", "Syncing...", false)
+}
+
+def refreshSyncStatus() {
+    def changes = pendingChanges
+    sendEventIfNew("syncStatus", (changes ?  "${changes} Pending Changes" : "Synced"), false)
+}
+
+private static getCommandClassVersions() {
+    [
+            0x20: 1,	// Basic
+            0x25: 1,	// Switch Binary
+            0x55: 1,	// Transport Service
+            0x59: 1,	// AssociationGrpInfo
+            0x5A: 1,	// DeviceResetLocally
+            0x27: 1,	// Switch All
+            0x5E: 2,	// ZwaveplusInfo
+            0x6C: 1,	// Supervision
+            0x70: 1,	// Configuration
+            0x7A: 2,	// FirmwareUpdateMd
+            0x72: 2,	// ManufacturerSpecific
+            0x73: 1,	// Powerlevel
+            0x85: 2,	// Association
+            0x86: 1,	// Version (2)
+            0x8E: 2,	// Multi Channel Association
+            0x98: 1,	// Security S0
+            0x9F: 1		// Security S2
+    ]
+}
+
+private getPendingChanges() {
+    return configParams.count { "${it.value}" != "${getParamStoredValue(it.num)}" }
+}
+
+private getParamStoredValue(paramNum) {
+    return safeToInt(state["configVal${paramNum}"] , null)
+}
+
+private setParamStoredValue(paramNum, value) {
+    state["configVal${paramNum}"] = value
+}
+
+private getConfigParams() {
+    return [
+            ledModeParam,
+            autoOffIntervalParam,
+            autoOnIntervalParam,
+            powerFailureRecoveryParam
+    ]
+}
+
+
+private getLedModeParam() {
+    return getParam(1, "LED Indicator Mode", 1, 0, ledModeOptions)
+}
+
+private getAutoOffIntervalParam() {
+    return getParam(2, "Auto Turn-Off Timer", 4, 0, autoOnOffIntervalOptions)
+}
+
+private getAutoOnIntervalParam() {
+    return getParam(4, "Auto Turn-On Timer", 4, 0, autoOnOffIntervalOptions)
+}
+
+private getPowerFailureRecoveryParam() {
+    return getParam(6, "Power Failure Recovery", 1, 0, powerFailureRecoveryOptions)
+}
+
+private getParam(num, name, size, defaultVal, options) {
+    def val = safeToInt((settings ? settings["configParam${num}"] : null), defaultVal)
+
+    def map = [num: num, name: name, size: size, value: val]
+    if (options) {
+        map.options = setDefaultOption(options, defaultVal)
+    }
+
+    return map
+}
+
+private static setDefaultOption(options, defaultVal) {
+    return options?.collectEntries { k, v ->
+        if ("${k}" == "${defaultVal}") {
+            v = "${v} [DEFAULT]"
+        }
+        ["$k": "$v"]
+    }
+}
+
+private sendEventIfNew(name, value, displayed=true, type=null, unit="") {
```

Comment:
I'm not sure about this method. Won't it filter out too many events? Like those which would be sent when device is offline, but since status technically don't change, they won't? Therefore device won't be brought back from being offline?

Reply:
I'm not sure too, so I keep it.

Reply:
I've meant "I'm not sure if this method should be implemented at all". 

On second thought - please remove it and just send events in normal way. Afaik Device API takes care of verifying whether status has changed, or not.

Reply:
Again, just send events without this method.

Reply:
OK, I replace it with sendEvent.